pkg/network/infraconfigurators/bridge.go

```diff
@@ -188,26 +180,32 @@ func (b *BridgePodNetworkConfigurator) GenerateDomainIfaceSpec() api.Interface {
 	}
 }
 
-func (b *BridgePodNetworkConfigurator) learnInterfaceRoutes() error {
+func (b *BridgePodNetworkConfigurator) learnInterfaceRoutes() ([]netlink.Route, error) {
 	routes, err := b.handler.RouteList(b.podNicLink, netlink.FAMILY_V4)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to get routes for %s", b.podNicLink.Attrs().Name)
-		return err
+		return nil, err
 	}
 	if len(routes) == 0 {
-		return fmt.Errorf("No gateway address found in routes for %s", b.podNicLink.Attrs().Name)
+		return nil, fmt.Errorf("No gateway address found in routes for %s", b.podNicLink.Attrs().Name)
 	}
-	b.podIfaceRoutes = routes
-	return nil
+	return routes, nil
 }
 
-func (b *BridgePodNetworkConfigurator) decorateDhcpConfigRoutes(dhcpConfig *cache.DHCPConfig) {
-	log.Log.V(4).Infof("the default route is: %s", b.podIfaceRoutes[0].String())
-	dhcpConfig.Gateway = b.podIfaceRoutes[0].Gw
-	if len(b.podIfaceRoutes) > 1 {
-		dhcpRoutes := netdriver.FilterPodNetworkRoutes(b.podIfaceRoutes, dhcpConfig)
-		dhcpConfig.Routes = &dhcpRoutes
+func (b *BridgePodNetworkConfigurator) decorateDhcpConfigRoutes(dhcpConfig *cache.DHCPConfig) error {
```

Comment:
imo, this now stopped being this a renderer of already available information; it now actively fetches the info.

I would rename the method accordingly.

Reply:
Removed the routes change from this commit to keep the separation between fetching the info and using it for generation.